dali/operators/random/noise/shot_noise.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/random/noise/shot_noise.h"
+#include "dali/operators/random/rng_base_cpu.h"
+#include "dali/pipeline/operator/arg_helper.h"
+
+namespace dali {
+
+DALI_SCHEMA(noise__Shot)
+    .DocStr(R"code(Applies shot noise to the input.
+
+The shot noise is generated by applying the following formula::
+
+    output[i] = denormalize(poisson_dist(normalize(input[:]) * peak) / peak)
+
+where ``normalize`` and ``denormalize`` represent scaling to the range [-1, 1] and back to
```

Comment:
Poisson noise is unsigned - if so, then why do we have the [-1,1] range? Also - do we want to do something with negative values? (e.g. produce a noisy value from the absolute and copysign?).